Sources/SmithyIdentityAPI/ClientConfigDefaultIdentityResolver.swift

```diff
@@ -0,0 +1,17 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+/// A protocol on identity resolver used to signify that this resolver is a default resolver created because the client config was not passed a custom resolver at creation.
+///
+/// Resolvers that do not implement this protocol should be presumed to not be a client config default.
+
```

Comment:
The protocol below is used to determine if a credential resolver came from client config defaults.  It's protected with `@_spi` so it isn't visible to customers.